### DISCUSSION TITLE
You can use Chat GPT to make Scratch extensions.

### DISCUSSION THREAD

**Discussion by TheShovel** [TurboWarp/extensions#310] [General]
Just a little thing I thought you guys would enjoy playing around with, in case you didn't know it was possible.
![image](https://user-images.githubusercontent.com/68913917/227963012-8ad04f2b-ad07-4f22-8fc9-343a6e8ecf03.png)

**Comment by kindpump** [comment]
I've tried this several times and haven't gotten any of them to actually work yet, even though they look like they would. Any suggestions on how to promt it to get a compatible extension

**Comment by TheShovel** [comment]
Try to tell it to use JavaScript. I noticed that it used another programming language sometimes.

**Comment by MA-master-sec** [comment]
It's not clever enough...
![image](https://user-images.githubusercontent.com/126303227/233758836-e0a75e17-472a-4520-a71f-4fbb5a8195e2.png)
It's obvious that it's wrong

**Reply by Flappy25** [reply]
I wish it was that easy

**Reply by MA-master-sec** [reply]
It's the extension belongs to ScratchX

**Reply by Sunny-da-moth** [reply]
if you first provide chatgpt with a working extension and then prompt it to make another extension then it will work better

**Comment by BlueDome77** [comment]
Tried with the same prompt and got something that works, though I had to change the block type from a command to reporter. Defiantly not the most efficient way to add two numbers together lol

![Screenshot 2023-07-19 at 1 12 01 PM](https://github.com/TurboWarp/extensions/assets/135789443/a48a2092-7e4b-46fa-b1f4-b6aa172e90ed)

**Comment by PURPL3-CAT** [comment]
If you explain how extensions are structured, and you copy and paste the articles from the TurboWarp documentation for making extensions, it can sort of understand how to make one.

**Comment by yuri-kiss** [comment]
it does not seem to use the right format XD

**Comment by 14-3dgar** [comment]
you have to send him an actual extension like one from the Custom extension tutorial from turbowarp [https://docs.turbowarp.org/development/extensions/hello-world](url)
Say its an example of a scratch extension and ask it to make another one with the things you want

**Reply by Flappy25** [reply]
Him?

**Reply by BlueDome77** [reply]
Him?

**Reply by Flappy25** [reply]
He addressed chatGPT as "him"

**Reply by BlueDome77** [reply]
yeah I know

**Reply by Flappy25** [reply]
Creepy that people are calling it a "him"

**Reply by 14-3dgar** [reply]
I'm not native english speaker. is it that bad telling chatgpt him? :/

**Reply by Flappy25** [reply]
It's just because it's a bit creepy to address a non human as "him"

**Reply by yuri-kiss** [reply]
👀
